Extensions/P2P/B_p2ptools.js

```diff
@@ -0,0 +1,308 @@
+ * @returns {string} - The data as JSON.
+ */
+gdjs.evtTools.p2p.getEventData = function(eventName) {
+  var dataLoss = gdjs.evtTools.p2p.eventHandling[eventName];
+  if (typeof dataLoss === "undefined" || dataLoss === false) {
+    var event = gdjs.evtTools.p2p.lastEventData[eventName];
+    return event[event.length - 1];
+  } else {
+    return gdjs.evtTools.p2p.lastEventData[eventName];
+  }
+}
+
+/**
+ * Get a variable associated to the last trigger of an event.
+ * @param {string} eventName - The event to get the variable from.
+ * @param {gdjs.Variable} variable - The variable where to store the variable content.
+ */
+gdjs.evtTools.p2p.getEventVariable = function(eventName, variable) {
+  gdjs.evtTools.network.jsonToVariableStructure(gdjs.evtTools.p2p.getEventData(eventName), variable);
+}
+
+/**
+ * Connects to a custom broker server.
+ * @param {string} host The host of the broker server
+ * @param {number} port
+ * @param {string} path The path (part of the url after the host) to the broker server
+ * @param {string} key Optional password to connect to the broker server
+ * @param {boolean} ssl Use ssl?
+ */
+gdjs.evtTools.p2p.useCustomBrokerServer = function(host, port, path, key, ssl) {
+  key = key.length === 0 ? "peerjs" : key; // All servers have "peerjs" as default key
+  gdjs.evtTools.p2p.peerConfig = {
+    debug: 1,
+    host,
+    port,
+    path,
+    secure: ssl,
+    key,
+  };
+  gdjs.evtTools.p2p.loadPeerJS();
+}
+
+/**
+ * Use default broker server. NOT RECOMMENDED.
+ */
+gdjs.evtTools.p2p.useDefaultBrokerServer = function() {
+  gdjs.evtTools.p2p.loadPeerJS();
+}
+
+/**
+ * Returns the own current peer ID
+ * @see Peer.id
+ * @returns {string}
+ */
+gdjs.evtTools.p2p.getCurrentId = function() {
+  if(gdjs.evtTools.p2p.peer == undefined) return "";
+  return gdjs.evtTools.p2p.peer.id || "";
+}
+
+/**
+ * Returns true once PeerJS is initialized
+ * @see gdjs.evtTools.p2p.ready
+ * @returns {boolean}
+ */
+gdjs.evtTools.p2p.isReady = function() {return gdjs.evtTools.p2p.ready;}
+
+/**
+ * Returns true once when there is an error.
+ * @returns {boolean}
+ */
+gdjs.evtTools.p2p.onError = function() {
+  var returnValue = gdjs.evtTools.p2p.error;
+  gdjs.evtTools.p2p.error = false;
+  return returnValue;
+}
+
+gdjs.evtTools.p2p.getLastError = function() {
+  return gdjs.evtTools.p2p.lastError;
+}
+
+/**
+ * Returns true once a peer disconnected.
+ * @returns {boolean}
+ */
+gdjs.evtTools.p2p.onDisconnect = function() {
+  var returnValue = gdjs.evtTools.p2p.peerDisconnected;
+  gdjs.evtTools.p2p.peerDisconnected = false;
+  return returnValue;
+}
+
+gdjs.evtTools.p2p.getDisconnectedPeer = function() {
+  return gdjs.evtTools.p2p.disconnectedPeer;
+}
+
+gdjs.callbacksRuntimeScenePostEvents.push(function() {
+  for(var i in gdjs.evtTools.p2p.lastEventData) {
+    if(typeof gdjs.evtTools.p2p.lastEventData[i] === "object" && gdjs.evtTools.p2p.lastEventData[i].length > 0)
+      gdjs.evtTools.p2p.lastEventData[i].length--;
```

Comment:
Let's use `pop` instead, I'm afraid manipulating the length of an array can deoptimize it on some engines.